### PR TITLE
improvements to path munging for cache_paths.

### PR DESCRIPTION
**Pull request by mmccarty**
I noticed a few rough edges when using http remote files.  This PR addresses them.

### CODE REVIEW

File: intake/source/cache.py [hunk 1]
```diff
@@ -8,11 +8,21 @@
...
 
 logger = logging.getLogger('intake')
 
 
+def strip_http(path):
+    storage_option = infer_storage_options(path)
+
+    protocol = storage_option['protocol']
+    if protocol in ('http', 'https'):
+        return path.replace("{}://".format(protocol), '')
```
Comment (martindurant) [inline intake/source/cache.py]: Hm, right... so most FSs remove the protocol, but http doesn't?
This needs a comment in the code at least.

File: intake/source/cache.py [hunk 2]
```diff
@@ -44,12 +54,17 @@ def _ensure_cache_dir(self):
 
     def _munge_path(self, cache_subdir, urlpath):
         import re
+
+        regex = os.path.normpath(strip_http(self._spec['regex']))
+        path = os.path.normpath(strip_http(urlpath))
```
Comment (martindurant) [inline intake/source/cache.py]: Is `normpath` necessary? What failed without it? Worry that only local FS it makes no difference, but remote FS might require original key without munge (or is this only for local files in the cache?)

File: intake/source/cache.py [hunk 3]
```diff
@@ -106,8 +121,14 @@ def load(self, urlpath):
             cache_path = file_out.path
             cache_paths.append(cache_path)
 
+            # If `_munge_path` did not find a match we want to avoid writing to the urlpath.
+            if cache_path == urlpath:
+                continue
+
             if not os.path.isfile(cache_path):
-                logger.info("Caching file {} from urlpath {}".format(file_in.path, urlpath))
+                logger.info("Caching file: {}".format(file_in.path))
```
Comment (martindurant) [inline intake/source/cache.py]: Change these to debug and add docs for how to set logging level.

File: intake/source/cache.py (hunk 1 above)
Comment (mmccarty) [reply intake/source/cache.py]: That's right.  I'll add a comment.

File: intake/source/cache.py (hunk 2 above)
Comment (mmccarty) [reply intake/source/cache.py]: I'm using it here to remove trailing slashes. :-)

File: intake/source/cache.py (hunk 3 above)
Comment (mmccarty) [reply intake/source/cache.py]: 👍

File: intake/source/cache.py (hunk 2 above)
Comment (mmccarty) [reply intake/source/cache.py]: Also, notice I'm calling `normpath` on the striped `regex` which http is the only case that will require the protocol in the regex to match properly.  However, you're right about the `urlpath`.  I'll fix that.

**Comment by mmccarty** [comment]
Any further comments? Or can we merge this one?

**Pull request merged**